src/lib/components/workflow-actions.svelte

```diff
@@ -18,20 +20,30 @@
   import Input from '$lib/holocene/input/input.svelte';
   import MenuDivider from '$lib/holocene/primitives/menu/menu-divider.svelte';
   import JSONEditor from '$lib/holocene/json-editor.svelte';
+  import { eventHistory } from '$lib/stores/events';
+  import { getFirstResetEventID } from '$lib/utilities/get-first-reset-event-id';
+  import { resetWorkflows } from '$lib/stores/reset-workflows';
+  import WorkflowResetForm from '$lib/components/workflow/workflow-reset-form.svelte';
 
   export let workflow: WorkflowExecution;
   export let namespace: string;
   export let cancelInProgress: boolean;
   export let terminateEnabled: boolean;
   export let cancelEnabled: boolean;
   export let signalEnabled: boolean;
+  export let resetEnabled: boolean;
 
   let reason = '';
   let signalInput = '';
   let signalName = '';
   let showTerminationConfirmation = false;
   let showCancellationConfirmation = false;
   let showSignalConfirmation = false;
+  let showResetConfirmation = false;
+  let resetType: ResetType = ResetType.FirstWorkflowTask;
```

Comment:
I like that idea in theory, but I think it requires some hackery to allow a component to bind to an Object rather than a primitive value. I tried it out and was not able to come up with an immediate solution, so I'm going to leave this as is for now.